File: kaola_manager_web/src/main/webapp/js/controller/brandController.js
app.controller("brandController",function ($scope,$http,$controller,brandService,$http) {
    $controller('baseController',{$scope:$scope})
    $scope.findAll=function () {
        //查询全部
        brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        )
    }

    //查询分页
    $scope.findPage=function (page,rows) {
       brandService.findPage(page,rows).success(
            function (response) {
                $scope.paginationConf.totalItems=response.total; //总条数
                $scope.list=response.rows;
            }
        )
    }
    //保存
    $scope.save=function () {
        //判断
        var method=null;
        var obj = null;
        if ($scope.entity.id==null){
            obj = brandService.save($scope.entity);
        } else {
            obj = brandService.update($scope.entity);
        }

       obj.success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();
                }else {
                    alert(response.message);
                }
            }
        )
    }
    //查询单条
    $scope.findOne=function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity=response;
            }
        )
    }



    //搜索
    $scope.searchEntity={};
    $scope.search=function (page,rows) {
        brandService.search(page,rows,$scope.searchEntity).success(
            function (response) {
                $scope.paginationConf.totalItems=response.total; //总条数
                $scope.list=response.rows;
            }
        )
    }


    //删除
    $scope.dele=function () {
        brandService.dele($scope.selectIds).success(
            function (response) {
                if (response.success){
                    $scope.reloadList();
                }else {
                    alert("删除失败");
                }
            }
        )
    }




})

File: kaola_manager_web/src/main/java/com/kaola/manager/controller/BrandController.java
package com.kaola.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.kaola.pojo.TbBrand;
import com.kaola.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;


    //查询品牌下拉列表
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }


    //查询全部
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }
    //分页查询
    @RequestMapping("/findPage")
    public PageResult findPage(int page,int rows){
        return brandService.findPage(page,rows);
    }
    //添加
    @RequestMapping("/save")
    public Result save(@RequestBody TbBrand tbBrand){
        try {
            brandService.save(tbBrand);
            return new Result(true,"保存成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"保存失败");
        }
    }
    //修改
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }
    //修改前的查询
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }
    //删除
    @RequestMapping("/del")
    public Result delet(Long[] ids){
        try {
            brandService.del(ids);
            return new Result(true,"删除成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }
    //条件搜索方法
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand,int page,int rows){
        PageResult pageResult = brandService.search(page, rows, brand);
        return pageResult;

    }
}
